cargo/config.go

```diff
@@ -69,6 +79,11 @@ func EncodeConfig(writer io.Writer, config Config) error {
 		return err
 	}
 
+	c, err = convertPatches(config.Metadata.DependencyConstraints, c)
```

Comment:
The Metadata can be technically empty/nil right?

Reply:
Yes but in `convertPatches`, if there is no metadata we just return `c`

Reply:
Even if `config.Metadata` is empty, we can still check `DependencyConstraints`